staging/src/k8s.io/legacy-cloud-providers/azure/clients/securitygroupclient/azure_securitygroupclient_test.go

```diff
@@ -102,6 +216,198 @@ func TestList(t *testing.T) {
 	assert.Equal(t, 3, len(result))
 }
 
+func TestListNextResultsMultiPages(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	reader1 := bytes.NewReader([]byte(`{"foo":"bar"}`))
+	reader1.Seek(13,0)
```

Comment:
@feiskyer I hack Reader.i to be 13 using Reader.Seek here, I am not sure I am doing right or wrong, but it make the test pass...

Reply:
Why do you set these two different readers?

Reply:
if I don't hack this, it will get below error for go test and 2 readers has different Reader.i values:
```
Error:          Not equal:
                                expected: network.SecurityGroupListResult{Response:autorest.Response{Response:(*http.Response)(0xc0000f9b90)}, Value:(*[]network.SecurityGroup)(nil), NextLink:(*string)(nil)}
                                actual  : network.SecurityGroupListResult{Response:autorest.Response{Response:(*http.Response)(0xc0000f9b00)}, Value:(*[]network.SecurityGroup)(nil), NextLink:(*string)(nil)}

                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -14,3 +14,3 @@
                                      },
                                -     i: (int64) 0,
                                +     i: (int64) 13,
                                      prevRune: (int) -1
```

Reply:
It's expected. Please ignore the reader.i when comparing it in the test code.

Reply:
thanks for clarifying! can you share me how to ignore reader.i as I am using below compare:
```
			response := &http.Response{
				StatusCode: test.statusCode,
			        Body:       test.body,
			}
			expected := network.SecurityGroupListResult{}
			expected.Response = autorest.Response{Response: response}
			assert.Equal(t, expected, result)
```

Reply:
it's a little hacky of seeking the reader, let's skip validations for the reader by removing L283

Reply:
will do it, thanks.